src/systems/idle-detector.js

```diff
@@ -17,48 +17,45 @@ const BASIC_ACTIVITY_PATHS = [
   paths.actions.angularVelocity
 ];
 
-AFRAME.registerSystem("idle-detector", {
-  init() {
-    this.resetTimeout = this.resetTimeout.bind(this);
-    this.idleTimeout = null;
-    this.lastInputCheck = 0;
+let idleTimeout = null;
+const lastInputCheck = 0;
 
-    const events = ["click", "pointerdown", "touchstart", "keyup"];
+const onIdleTimeout = () => {
+  window.dispatchEvent(new CustomEvent("idle_detected"));
```

Comment:
One of the things we are trying to get out of our code is event dispatching. It makes the code flow harder to follow. 

I am not super familiar with this code but tracing this even it looks like the only thing using this (and the other event we are firing below) is the code to show the timeout UI. 

Essentially this code https://github.com/mozilla/hubs/blob/cd8e2a12d50126148f94d14d790dcb3db23eaad9/src/react-components/ui-root.js#L510-L536 is really part of what "idle-detector" is doing at its core and it would be nice if that was more obvious... Unfortunately we don't really have a nice clean way for communicating between our game code and react UI code, so I am not sure how I would go about untangling it. The countdown and exiting logic could be moved but not sure how I would go about showing/updating the UI or capturing clicking the cancel button...

I don't think this is particularly important, the code works fine and its not integral to main gameplay, but again I think porting systems gives us the opportunity to rethink problems and clean up our tech debt.